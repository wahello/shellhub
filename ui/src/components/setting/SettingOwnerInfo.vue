<template>
  <div
    v-if="!isOwner"
    style="text-align:center"
  >
    <div data-test="message-div">
      <h3 class="pl-6">
        <span>
          You're not the owner of this namespace.
        </span>

        <p data-test="contactUser-p">
          Contact {{ namespaceOwnerName() }} user for more information.
        </p>
      </h3>
    </div>
    <br>
  </div>
</template>

<script>

export default {
  props: {
    isOwner: {
      type: Boolean,
      required: true,
    },
  },

  computed: {
    namespace() {
      return this.$store.getters['namespaces/get'];
    },

    owner() {
      return this.$store.getters['namespaces/get'].owner;
    },
  },

  methods: {
    namespaceOwnerName() {
      if (this.namespace.members !== undefined) {
        return this.namespace.members.find((x) => x.id === this.owner).name;
      }
      return null;
    },
  },
};

</script>
