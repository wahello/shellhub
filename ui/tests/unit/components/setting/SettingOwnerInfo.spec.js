import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import SettingOwnerInfo from '@/components/setting/SettingOwnerInfo';

describe('SettingOwnerInfo', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);
  const vuetify = new Vuetify();

  let wrapper;

  const isOwner = true;

  const namespace = {
    name: 'namespace1',
    owner: '124',
    members: [{ id: '124', name: 'user4' }, { id: '123', name: 'user1' }, { id: '125', name: 'user5' }],
    tenant_id: 'a736a52b-5777-4f92-b0b8-e359bf484713',
  };

  const store = new Vuex.Store({
    namespaced: true,
    state: {
      namespace,
    },
    getters: {
      'namespaces/get': (state) => state.namespace,
    },
    actions: {
    },
  });

  ///////
  // In this case, when the user owns the namespace and the focus of
  // the test is to check if the message no exists.
  ///////

  describe('Owner is true', () => {
    beforeEach(() => {
      wrapper = shallowMount(SettingOwnerInfo, {
        store,
        localVue,
        propsData: { isOwner },
        vuetify,
        stubs: ['fragment'],
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue intance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    //////
    // HTML validation
    //////

    it('Renders the template with data', () => {
      expect(wrapper.find('[data-test="message-div"]').exists()).toBe(false);
    });
  });

  ///////
  // In this case, when the user owns the namespace and the focus of
  // the test is to check if the message exists.
  ///////

  describe('Owner is false', () => {
    beforeEach(() => {
      wrapper = shallowMount(SettingOwnerInfo, {
        store,
        localVue,
        propsData: { isOwner: !isOwner },
        vuetify,
        stubs: ['fragment'],
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue intance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    //////
    // HTML validation
    //////

    it('Renders the template with data', () => {
      expect(wrapper.find('[data-test="message-div"]').exists()).toBe(true);

      const namespaceOwnerMessage = `Contact ${namespace.members[0].name} user for more information.`;
      expect(wrapper.find('[data-test=contactUser-p]').text()).toEqual(namespaceOwnerMessage);
    });
  });
});
