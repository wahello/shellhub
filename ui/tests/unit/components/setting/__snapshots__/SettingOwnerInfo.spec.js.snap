// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingOwnerInfo Owner is false Renders the component 1`] = `
<div style="text-align: center;">
  <div data-test="message-div">
    <h3 class="pl-6"><span>
        You're not the owner of this namespace.
      </span>
      <p data-test="contactUser-p">
        Contact user4 user for more information.
      </p>
    </h3>
  </div> <br>
</div>
`;

exports[`SettingOwnerInfo Owner is true Renders the component 1`] = `""`;
